src/transformers/models/mask2former/modeling_mask2former.py

```diff
@@ -843,6 +870,51 @@ def multi_scale_deformable_attention(
     )
     return output.transpose(1, 2).contiguous()
 
+class Mask2Former3DSinePositionEmbedding(nn.Module):
+    """
+    This is a more standard version of the position embedding, very similar to the one used by the Attention is all you
+    need paper, generalized to work on images.
+    """
+
+    def __init__(
+        self, num_pos_feats: int = 64, temperature: int = 10000, normalize: bool = False, scale: Optional[float] = None
+    ):
+        super().__init__()
+        if scale is not None and normalize is False:
+            raise ValueError("normalize should be True if scale is passed")
+        self.num_pos_feats = num_pos_feats
+        self.temperature = temperature
+        self.normalize = normalize
+        self.scale = 2 * math.pi if scale is None else scale
+
+    def forward(self, x: Tensor, mask: Optional[Tensor] = None) -> Tensor:
```

Comment:
It's preferable to use more descriptive argument names:
```suggestion
    def forward(self,  hidden_states: Tensor, mask: Optional[Tensor] = None) -> Tensor:
```